RecoLocalFastTime/FTLClusterizer/interface/BTLRecHitsErrorEstimatorIM.h

```diff
@@ -0,0 +1,43 @@
+#ifndef RecoLocalFastTime_FTLClusterizer_BTLRecHitsErrorEstimatorIM_H
+#define RecoLocalFastTime_FTLClusterizer_BTLRecHitsErrorEstimatorIM_H 1
+
+//-----------------------------------------------------------------------------
+// \class         BTLRecHitsErrorEstimatorIM
+// Used to improve the local error of recHits and TrackingrecHits in BTL
+//-----------------------------------------------------------------------------
+
+#include "Geometry/MTDGeometryBuilder/interface/MTDGeomDetUnit.h"
+#include "Geometry/MTDGeometryBuilder/interface/RectangularMTDTopology.h"
+#include "Geometry/MTDGeometryBuilder/interface/ProxyMTDTopology.h"
+
+#include "Geometry/CommonDetUnit/interface/GeomDetEnumerators.h"
+
+#include "Geometry/MTDGeometryBuilder/interface/MTDGeometry.h"
+
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+class BTLRecHitsErrorEstimatorIM {
+public:
+  BTLRecHitsErrorEstimatorIM(const MTDGeomDet* det, const LocalPoint& lp) : det_(det), lp_(lp) {}
+  LocalError localError() const {
+    if (GeomDetEnumerators::isEndcap(det_->type().subDetector())) {
+      throw cms::Exception("BTLRecHitsErrorEstimatorIM")
+          << "This is an object from Endcap. Only use it for the Barrel!" << std::endl;
+      return LocalError(0, 0, 0);
+    }
+    const ProxyMTDTopology& topoproxy = static_cast<const ProxyMTDTopology&>(det_->topology());
+    const RectangularMTDTopology& topo = static_cast<const RectangularMTDTopology&>(topoproxy.specificTopology());
+    MeasurementPoint mp = topo.measurementPosition(lp_);
+    MeasurementError simpleRect(1. / 12., 0, 1. / 12.);
+    LocalError error_before = topo.localError(mp, simpleRect);
+    LocalError error_modified(error_before.xx(), error_before.xy(), 0.36);
```

Comment:
Hi,
0.36 comes from our position error measurement, which is `positionError*positionError = 0.6*0.6 = 0.36`
I am sorry that it's like a hard coded way in the code. But do you have any suggestions about how I can pass it into the class elegantly?

Thanks.

Reply:
It sounds like at this point this is a non-trivially derived quantity.
It at least needs a comment with a reference where it came from.
Eventually this should become a part of conditions data.
Due to (s)low support to condFormats schema evolution, this may be pushed over for later for some time.
I guess for now it should be OK to have this value configurable.

Reply:
Hi,
Just to make sure, I should just keep the code and add some comment explaining why do we use 0.36 right?
Thanks.

Reply:
a configurable parameter would be better than a hardcoded value

Reply:
Cool I will try to do that.
Thanks!

Reply:
> a configurable parameter would be better than a hardcoded value

Hi,
I am sorry that I am not very clear about making the parameter in the class configurable. I know that some classes that get called in a python file can have configure parameters. But this class is just called inside the Clusterizer and MTDCPEFromSiPM class. Could you give me some instructions of examples to make class like that have configurable parameters? 
Thanks.

Reply:
OK, it sounds simpler to just add a clear note where this 0.6^2 came from.